libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -79,7 +81,7 @@ type
   PubSubPeerEvent* = object
     kind*: PubSubPeerEventKind
 
-  GetConn* = proc(): Future[Connection] {.gcsafe, raises: [].}
+  GetConn* = proc(): Future[Connection] {.gcsafe, async: (raises: [GetConnDialError]).}
```

Comment:
`gcsafe` is not needed either, with `async`

for some callbacks, it's worth considering allowing `CancelledError` - again, depending on the context, I don't remember this one in particular, but "cancelling a connection attempt" seems reasonable logically at least